src/app/data/services/local_storage_service.ts

```diff
@@ -6,17 +6,24 @@ class LocalStorageService {
   private expiryDate:{[key:string]:number} = {};
   private serializer:AjaxContentBlocksHashSerializer = new AjaxContentBlocksHashSerializer();
 
+  localStorage:Storage;
+  constructor(storage:Storage) {
+    this.localStorage = storage;
+  }
+
   fetch(key:string):AjaxContentBlocksHash {
-    const result = this.serializer.parse(localStorage.getItem(key));
+    if (!this.localStorage) { return {}; }
+    const result = this.serializer.parse(this.localStorage.getItem(key));
     return result ? result : {};
   }
 
   put(key:string, hash:AjaxContentBlocksHash) {
+    if (!this.localStorage) return;
```

Comment:
this was in a `try` anyways. So we could still show the error in the console. what do you think?